Entity/Comment.php

```diff
@@ -134,6 +141,37 @@ public function setThread(ThreadInterface $thread): CommentInterface
         return $this;
     }
 
+    public function getParent(): ?CommentInterface
+    {
+        return $this->parent;
+    }
+
+    public function setParent(?CommentInterface $parent = null): CommentInterface
+    {
+        $this->parent = $parent;
+
+        return $this;
+    }
+
+    public function getDepth(): int
+    {
+        return $this->depth;
+    }
+
+    public function getComments(): Collection
+    {
+        return $this->children;
+    }
+
+    public function getPublishedComments(): Collection
```

Comment:
same as above

Reply:
i have no big opinion about this - we can call it `getChildren()` and `getPublishedChildren()`